pkg/tfvars/ovirt/ovirt.go

```diff
@@ -71,6 +78,50 @@ func TFVars(
 		}
 		cfg.BaseImageLocalFilePath = imageFilePath
 	}
-
+	if len(controlPlaneAffinityGroups) > 0 {
+		cfg.ControlPlaneAffinityGroups = createControlPlaneAffinityGroups(controlPlaneAffinityGroups, infraID)
+	}
+	cfg.ComputeAffinityGroups = createComputeAffinityGroups(
+		computeMachines, controlPlaneAffinityGroups, infraID)
 	return json.MarshalIndent(cfg, "", "  ")
 }
+
+func createControlPlaneAffinityGroups(masterAffiniryGroups []*ovirt.AffinityGroup, infraID string) []map[string]interface{} {
+	var agMap []map[string]interface{}
+	for _, ag := range masterAffiniryGroups {
+		if ag.Name != "" {
+			agMap = append(agMap, map[string]interface{}{
+				"name":     strings.Replace(ag.Name, ovirtdefaults.DAffinyGroupBase, infraID, 1),
+				"priority": float64(ag.Priority),
+			})
+		}
+	}
+	return agMap
+}
+
+func createComputeAffinityGroups(computeMachines []*ovirt.MachinePool, masterAffiniryGroups []*ovirt.AffinityGroup, infraID string) []map[string]interface{} {
+	var agMap []map[string]interface{}
+	uniqeAGName := make(map[string]*ovirt.AffinityGroup)
+
+	for _, pool := range computeMachines {
+		for _, poolAG := range pool.AffinityGroups {
+			if poolAG.Name != "" {
+				uniqeAGName[poolAG.Name] = poolAG
+			}
+		}
+	}
+	// If the affinity group exist on master list don't create it twice
```

Comment:
How are we going to deal with the affinity group name used for both control plane and compute but using different priorities for each? Is that going to be caught by validation?

Reply:
I will add this to the validations